extensions/vertx-http/deployment/src/main/java/io/quarkus/devui/deployment/menu/WorkspaceProcessor.java

```diff
@@ -0,0 +1,251 @@
+                return new WorkspaceActionResult(convertedPath, actionToExecute.getFunction().apply(t));
+            }
+            return null;
+        });
+
+        buildItemActions.addAction("getWorkspaceItemContent", (Map<String, String> params) -> {
+            if (params.containsKey("path")) {
+                Path path = Paths.get(URI.create(params.get("path")));
+                return readContents(path);
+            }
+            return null;
+        });
+
+        buildItemActions.addAction("saveWorkspaceItemContent", (Map<String, String> params) -> {
+            if (params.containsKey("content")) {
+                String content = params.get("content");
+                Path path = Paths.get(URI.create(params.get("path")));
+                writeContent(path, content);
+                return new SavedResult(path, true, null);
+            }
+            return new SavedResult(null, false, "Invalid input");
+        });
+
+        buildTimeActionProducer.produce(buildItemActions);
+    }
+
+    private void sortWorkspaceItems(List<WorkspaceBuildItem.WorkspaceItem> items) {
+        items.sort(Comparator.comparing((WorkspaceBuildItem.WorkspaceItem item) -> isFileInRoot(item.name()))
+                .thenComparing(item -> folderPriority(item.name()))
+                .thenComparing(WorkspaceBuildItem.WorkspaceItem::name));
+    }
+
+    private static int folderPriority(String name) {
+        if (name.startsWith("src/main/java"))
+            return 1;
+        if (name.startsWith("src/main/resources"))
+            return 2;
+        if (name.startsWith("src/main/"))
+            return 3;
+        if (name.startsWith("src/test/java"))
+            return 4;
+        if (name.startsWith("src/test/resources"))
+            return 5;
+        if (name.startsWith("src/test/"))
+            return 6;
+        if (name.startsWith("src/integrationTest/"))
+            return 7;
+        return 8;
+    }
+
+    private boolean isFileInRoot(String name) {
+        return !name.contains("/");
+    }
+
+    private String writeContent(Path path, String contents) {
+        try {
+            Files.createDirectories(path.getParent());
+            if (!Files.exists(path))
+                Files.createFile(path);
+            Files.writeString(path, contents, StandardOpenOption.TRUNCATE_EXISTING,
+                    StandardOpenOption.CREATE);
+            return path.toString();
+        } catch (IOException ex) {
+            throw new UncheckedIOException(ex);
+        }
+    }
+
+    private WorkspaceContent readContents(Path filePath) {
+        if (filePath != null && Files.exists(filePath) && Files.isReadable(filePath)) {
+            try {
+                String mimeType = Files.probeContentType(filePath);
+                if (mimeType == null)
+                    mimeType = getFileTypeManually(filePath);
+
+                if (mimeType.startsWith("image")) {
```

Comment:
That might be a bit limited.
We may have to do the opposite and list the mime type for text we know and do binary for everything else.